packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -0,0 +1,1235 @@
+
+// Examples can assume:
+// class MyDataObject {}
+
+/// A callback used by [ReorderableList] to report that a list item has moved
+/// to a new position in the list.
+///
+/// Implementations should remove the corresponding list item at [oldIndex]
+/// and reinsert it at [newIndex].
+///
+/// If [oldIndex] is before [newIndex], removing the item at [oldIndex] from the
+/// list will reduce the list's length by one. Implementations will need to
+/// account for this when inserting before [newIndex].
+///
+/// {@youtube 560 315 https://www.youtube.com/watch?v=3fB1mxOsqJE}
+///
+/// {@tool snippet}
+///
+/// ```dart
+/// final List<MyDataObject> backingList = <MyDataObject>[/* ... */];
+///
+/// void handleReorder(int oldIndex, int newIndex) {
+///   if (oldIndex < newIndex) {
+///     // removing the item at oldIndex will shorten the list by 1.
+///     newIndex -= 1;
+///   }
+///   final MyDataObject element = backingList.removeAt(oldIndex);
+///   backingList.insert(newIndex, element);
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [ReorderableList], a widget list that allows the user to reorder
+///    its items.
+///  * [SliverReorderableList], a sliver list that allows the user to reorder
+///    its items.
+///  * [ReorderableListView], a material design list that allows the user to
+///    reorder its items.
+typedef ReorderCallback = void Function(int oldIndex, int newIndex);
+
+/// Signature for the builder callback used to decorate the dragging item in
+/// [ReorderableList] and [SliverReorderableList].
+typedef ReorderItemProxyDecorator = Widget Function(Widget child, int index, Animation<double> animation);
+
+/// A scrolling container that allows the user to interactively reorder the
+/// list items.
+///
+/// This widget is similar to one created by [ListView.builder], and uses
+/// an [IndexedWidgetBuilder] to create each item.
+///
+/// It is up to the application to wrap each child (or an internal part of the
+/// child such as a drag handle) with a drag listener that will recognize
+/// the start of an item drag and then start the reorder by calling
+/// [ReorderableListState.startItemDragReorder].
+///
+/// This is most easily achieved by wrapping each child in a
+/// [ReorderableDragStartListener] or a [ReorderableDelayedDragStartListener].
+/// These will take care of recognizing the start of a drag gesture and call
+/// the list state's start item drag method.
+///
+/// This widget's [ReorderableListState] can be used manually start a item
+/// reorder, or cancel a current drag. To refer to the
+/// [ReorderableListState] either provide a [GlobalKey] or use the static
+/// [ReorderableList.of] method from an item's build method.
+///
+/// See also:
+///
+///  * [SliverReorderableList], a sliver list that allows the user to reorder
+///    its items.
+///  * [ReorderableListView], a material design list that allows the user to
+///    reorder its items.
+class ReorderableList extends StatefulWidget {
```

Comment:
I can't seem to figure out where `ReorderableList` is used. It seems like `SliverReorderableList` is what is used by `ReorderableListView`.

Reply:
`ReorderableList` is not used directly by anything in the framework. It is meant to be a non-opinionated base version of the reorderable list as a widget. If people don't want the built-in drag handles and decorations of the  Material `ReorderableListView` they can use this instead. Otherwise, yeah everything is just using the sliver version internally.